src/Handler/ErrorHandler/ErrorHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Mollie       https://www.mollie.nl
+ *
+ * @author      Mollie B.V. <info@mollie.nl>
+ * @copyright   Mollie B.V.
+ *
+ * @see        https://github.com/mollie/PrestaShop
+ *
+ * @license     https://github.com/mollie/PrestaShop/blob/master/LICENSE.md
+ * @codingStandardsIgnoreStart
+ */
+
+namespace Mollie\Handler\ErrorHandler;
+
+use Module;
+use Mollie;
+use Mollie\Config\Config;
+use MolliePrefix\Raven_Client;
+
+/**
+ * Handle Error.
+ */
+class ErrorHandler
+{
+	/**
+	 * @var Raven_Client
+	 */
+	protected $client;
+
+	/**
+	 * @var ErrorHandler
+	 */
+	private static $instance;
+
+	public function __construct()
+	{
+		/** @var Mollie */
+		$module = Module::getInstanceByName('mollie');
+
+		$this->client = new Raven_Client(
```

Comment:
You can also decouple our code by using ErrorHandlerClientInterface which instantiates Raven_Client and injecting it into constructor.
That would allow us to use different error handlers if we ever decide to.

Reply:
This method is used in most of prestashop modules, it allows to track exceptions everywhere without adding it each time to constructor. Because it gets called only on exceptions when it shouldn't even happen i think this is better then adding it to each constructor that wants to use sentry.

Reply:
But to test this code then i would have to create adapter and inject it into constructor anyways